website/server/libs/cron.js

```diff
@@ -130,6 +130,14 @@ function trackCronAnalytics (analytics, user, _progress, options) {
 
 function awardLoginIncentives (user) {
   if (user.loginIncentives > 50) return;
+
+  //  Remove old notifications if they exists
+  user.notifications
+    .toObject()
+    .find((notif, index) => {
```

Comment:
If you don't want it to exit early, why are you using find? You should be using forEach instead. 

Also, why not exit early? If this is working as intended, there should just be the one login incentives notification. The only exception is how it is currently, where we've allowed multiples to stack up. But that should be resolved as soon as a client accepts them. 

I could see the argument to keep it like this for a week to prune most of the duplicates, then switch to a version where it exits early.